vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -389,7 +392,16 @@ and
                 projectContext.RemoveSourceFile(file)
                 updated <- true
         
-        let updatedRefs = site.AssemblyReferences() |> wellFormedFilePathSetIgnoreCase
+        // Update the output assembly path projectContext.
+        let updatedBinOutputPath = site.CompilationBinOutputPath |> Option.toObj
+        let originalBinOutputPath = projectContext.BinOutputPath
+        if updatedBinOutputPath <> originalBinOutputPath then 
+            projectContext.BinOutputPath <- updatedBinOutputPath
+            updated <- true
+
+        // Update the metadata/project references in projectContext. Note that AbstractProject.cs converts metadata references
+        // to project references automagically by consulting the project tracker.
```

Comment:
What if the referenced project hasn't been setup yet?

Reply:
> What if the referenced project hasn't been setup yet?

TBH I don't fully understand the Roslyn code that does the metadata --> project reference conversion, but my impression is that it would handle this case.  Certainly the project references were in place on initial project load - which must encounter this